.github/workflows/ansible-test.yml

```diff
@@ -1,93 +1,176 @@
 
+      # ansible-test requires the collection to be in a directory in the form
+      # .../ansible_collections/NAMESPACE/COLLECTION_NAME/
+
       - name: Check out code
-        uses: actions/checkout@v1
+        uses: actions/checkout@v2
         with:
-          # FIXME ansible_collections/NAMESPACE/COLLECTION
-          path: ansible_collections/community/FIXME
+          path: ansible_collections/NAMESPACE/COLLECTION_NAME
 
-      - name: Set up Python 3.6
-        uses: actions/setup-python@v1
+      - name: Set up Python ${{ matrix.ansible }}
+        uses: actions/setup-python@v2
         with:
-          python-version: 3.6
+          python-version: ${{ matrix.python }}
 
+      # Install the head of the given branch (devel, stable-2.10)
       - name: Install ansible-base (${{ matrix.ansible }})
         run: pip install https://github.com/ansible/ansible/archive/${{ matrix.ansible }}.tar.gz --disable-pip-version-check
 
+      # run ansible-test sanity inside of Docker.
+      # The docker container has all the pinned dependencies that are required.
+      # Explicity specify the version of Python we want to test
       - name: Run sanity tests
-        run: ansible-test sanity --docker -v --color --python 3.6
+        run: ansible-test sanity --docker -v --color --python ${{ matrix.python }}
+        working-directory: ./ansible_collections/NAMESPACE/COLLECTION_NAME
+
+###
+# Unit tests (OPTIONAL)
+#
+# https://docs.ansible.com/ansible/latest/dev_guide/testing_units.html
 
-# If you don't have unit tests, you can delete this section
   units:
     runs-on: ubuntu-latest
+    name: Units (Ⓐ${{ matrix.ansible }}+py${{ matrix.python }}})
+    strategy:
+      fail-fast: false
+      matrix:
+        ansible:
+          # - stable-2.9 # Only if your collection supports Ansible 2.9
+          - stable-2.10
+          - devel
+        python:
+          - 2.7
+          - 3.7
+        exclude:
+          - python: 3.8  # blocked by ansible/ansible#70155
+
     steps:
       - name: Check out code
-        uses: actions/checkout@v1
+        uses: actions/checkout@v2
         with:
-          # FIXME ansible_collections/NAMESPACE/COLLECTION        
-          path: ansible_collections/community/grafana
+          path: ansible_collections/NAMESPACE/COLLECTION_NAME
 
-      - name: Set up Python 3.6
-        uses: actions/setup-python@v1
+      - name: Set up Python ${{ matrix.ansible }}
+        uses: actions/setup-python@v2
         with:
-          python-version: 3.6
+          python-version: ${{ matrix.python }}
+
+      - name: Install ansible-base (${{ matrix.ansible }})
+        run: pip install https://github.com/ansible/ansible/archive/${{ matrix.ansible }}.tar.gz --disable-pip-version-check
 
-      - name: Install ansible-base (devel)
-        run: pip install https://github.com/ansible/ansible/archive/devel.tar.gz --disable-pip-version-check
+      # OPTIONAL If your unit test requires Python libraries from other collections
```

Comment:
I'd rather recommend to put them in as dependencies instead. Unless they are optional.

Reply:
Dependencies how?
I don't believe you can build and install the collection then run `ansible-test`

Reply:
If you build and install the collection, it should install all the _runtime_ dependencies. So you need to have a separate location for the _test_ dependencies. Now, you suggest installing them using the CLI command and passing such deps there.
This will effectively make the GHA workflow config a place where those _test_ deps are listed.
There's no tooling to extract such deps for using elsewhere so this would force people to maintain copy-pasted lists of such deps in multiple places and not forget to update every each of them on any changes.
OTOH you could avoid this by introducing a centralized location for listing test deps, for example, `test-requirements.yml` file. Then, the only thing left to do would be to integrate an installation command that would install things from that file per 
https://docs.ansible.com/ansible/devel/user_guide/collections_using.html#install-multiple-collections-with-a-requirements-file.
This would allow integrating  the test deps into any other system by just doing the same, not having to worry about synchronizing the lists of deps.
https://github.com/ansible-collections/collection_template/pull/5/files#r452271854